tests/org.eclipse.papyrus.uml.interaction.model.tests/src/org/eclipse/papyrus/uml/interaction/model/tests/creation/CreateMessageTest.java

```diff
@@ -0,0 +1,196 @@
+import org.eclipse.papyrus.uml.interaction.tests.rules.ModelFixture;
+import org.eclipse.papyrus.uml.interaction.tests.rules.ModelResource;
+import org.eclipse.uml2.uml.Message;
+import org.eclipse.uml2.uml.MessageSort;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+@SuppressWarnings("restriction")
+@ModelResource({"CreateMessageTesting.uml", "CreateMessageTesting.notation" })
+public class CreateMessageTest {
+
+	@Rule
+	public final ModelFixture.Edit model = new ModelFixture.Edit();
+
+	private MInteraction interaction;
+
+	private int lifeline1Top;
+
+	private int lifeline2Top;
+
+	private int lifeline3Top;
+
+	private int message1Top;
+
+	private int message2Top;
+
+	private int execution1Top;
+
+	private int execution2Top;
+
+	private int execution3Top;
+
+	private int lifeline2Header;
+
+	private int lifeline3Header;
+
+	private int message3Top;
+
+	@Before
+	public void before() {
+		lifeline1Top = interaction().getLifelines().get(0).getTop().getAsInt();
+		lifeline2Top = interaction().getLifelines().get(1).getTop().getAsInt();
+		lifeline3Top = interaction().getLifelines().get(2).getTop().getAsInt();
+
+		lifeline2Header = getLifelineBodyTop(interaction().getLifelines().get(1)) - lifeline2Top;
+		lifeline3Header = getLifelineBodyTop(interaction().getLifelines().get(2)) - lifeline3Top;
+
+		message1Top = interaction().getMessages().get(0).getTop().getAsInt();
+		message2Top = interaction().getMessages().get(1).getTop().getAsInt();
+		if (interaction().getMessages().size() == 3) {
+			message3Top = interaction().getMessages().get(2).getTop().getAsInt();
+		}
+
+		execution1Top = interaction().getLifelines().get(0).getExecutionOccurrences().get(0).getTop()
+				.getAsInt();
+		execution2Top = interaction().getLifelines().get(1).getExecutionOccurrences().get(0).getTop()
+				.getAsInt();
+		execution3Top = interaction().getLifelines().get(2).getExecutionOccurrences().get(0).getTop()
+				.getAsInt();
+	}
+
+	private DiagramHelper diagramHelper() {
+		return LogicalModelPlugin.getInstance().getDiagramHelper(model.getEditingDomain());
+	}
+
+	private LayoutHelper layoutHelper() {
+		return LogicalModelPlugin.getInstance().getLayoutHelper(model.getEditingDomain());
+	}
+
+	private int getLifelineBodyTop(MLifeline lifeline) {
+		View shape = lifeline.getDiagramView().orElse(null);
+		return layoutHelper()//
+				.getTop(diagramHelper().getLifelineBodyShape(shape));
+	}
+
+	private MInteraction interaction() {
+		if (interaction == null) {
+			interaction = MInteraction.getInstance(model.getInteraction(), model.getSequenceDiagram().get());
+		}
+		return interaction;
+	}
+
+	private void execute(Command remove) {
+		if (!remove.canExecute()) {
+			Assert.fail("Command not executable"); //$NON-NLS-1$
+		}
+		remove.execute();
+		/* force reinit after change */
+		interaction = null;
+	}
+
+	// TODO tests for non executable inserts
```

Comment:
Please open a new issue for that instead of this TODO. I guess, you'd be the best person to address this new issue, right? If yes, please assign it to yourself.

Reply:
I opened up #334 and assigned to myself. 
I have not added it to any milestone. I think those tests are blocked until we decided on the cases where we do not allow the creation of certain messages. I will create a separate issue for this where I will explain the current (rather restrict approach) and explain the implications of the req suggested in #46. I will also include the cases for #44